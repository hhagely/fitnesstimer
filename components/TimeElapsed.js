import React, { Component } from 'react';
import { View, Text } from 'react-native';
import PropTypes from 'prop-types';

const leftPad = (width, n) => {
	if ((n + '').length > width) {
		return n;
	}
	const padding = new Array(width).join('0');
	return (padding + n).slice(-width);
};

class TimeElapsed extends Component {
	getUnits() {
		const { timeElapsed, isRunning, timerSettings } = this.props;

		const seconds =
			!isRunning && timeElapsed === 0
				? timerSettings.timerDuration * 60
				: timeElapsed / 1000;

		return {
			min: Math.floor(seconds / 60).toString(),
			sec: Math.floor(seconds % 60).toString(),
			msec: (seconds % 1).toFixed(1).substring(2)
		};
	}

	render() {
		const units = this.getUnits();
		return (
			<View>
				<Text style={{ fontSize: 108, textAlign: 'center' }}>
					{leftPad(2, units.min)}:{leftPad(2, units.sec)}
				</Text>
			</View>
		);
	}
}

TimeElapsed.propTypes = {
	timerSettings: PropTypes.shape({
		timerType: PropTypes.string,
		timerDuration: PropTypes.number,
		countdown: PropTypes.bool,
		countdownDuration: PropTypes.number,
		emomStyle: PropTypes.number
	})
};

export default TimeElapsed;
