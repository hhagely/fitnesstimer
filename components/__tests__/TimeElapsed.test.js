import 'react-native';
import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import toJson from 'enzyme-to-json';

import TimeElapsed from '../TimeElapsed';

configure({ adapter: new Adapter() });

describe('<TimeElapsed />', () => {
	it('renders the component', () => {
		const tree = shallow(<TimeElapsed />);

		expect(toJson(tree)).toMatchSnapshot();
	});
});
