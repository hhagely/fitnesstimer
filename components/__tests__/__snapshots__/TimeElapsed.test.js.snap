// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TimeElapsed /> renders the component 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 108,
        "textAlign": "center",
      }
    }
  >
    NaN
    :
    NaN
  </Text>
</View>
`;
